src/pages/organize/[orgId]/projects/activities/index.tsx

```diff
@@ -63,46 +65,53 @@ const CampaignActivitiesPage: PageWithLayout<CampaignActivitiesPageProps> = ({
   const onSearchStringChange = (evt: ChangeEvent<HTMLInputElement>) =>
     setSearchString(evt.target.value);
 
-  const activities = model.getStandaloneActivities().data;
-  const hasActivities = Array.isArray(activities) && activities.length > 0;
-
-  const activityTypes = activities?.map((activity) => activity.kind);
-  const filterTypes = [...new Set(activityTypes)];
-
   if (onServer) {
     return null;
   }
 
   return (
     <Box>
-      {!hasActivities && (
-        <ZUIEmptyState
-          href={`/organize/${orgId}/projects`}
-          linkMessage={messages.allProjects.linkToSummary()}
-          message={messages.allProjects.noActivities()}
-        />
-      )}
-      {hasActivities && (
-        <Grid container spacing={2}>
-          <Grid item sm={8}>
-            <ActivityList
-              allActivities={activities}
-              filters={filters}
-              orgId={parseInt(orgId)}
-              searchString={searchString}
-            />
-          </Grid>
-          <Grid item sm={4}>
-            <FilterActivities
-              filters={filters}
-              filterTypes={filterTypes}
-              onFiltersChange={onFiltersChange}
-              onSearchStringChange={onSearchStringChange}
-              value={searchString}
-            />
-          </Grid>
-        </Grid>
-      )}
+      <ZUIFuture future={model.getStandaloneActivities()} skeletonWidth={200}>
+        {(data) => {
+          if (data && data.length === 0) {
```

Comment:
You shouldn't have to do `data && data.length == 0`, because `ZUIFuture` should always return a non-null value. Does TypeScript complain when you do just `if (data.length == 0)`?

Reply:
Good point! TypeScript didn´t complained so I will just check the length in the condition!